Flaky-test: PrometheusMetricsTest.testBrokerConnection

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Example failure

https://github.com/coderzc/pulsar/actions/runs/3494482969/jobs/5850297404#step:8:1932

### Exception stacktrace

<!-- optionally provide the full stacktrace -->

<details>
<summary>Full exception stacktrace</summary>
<pre><code>
    Error:  testBrokerConnection(org.apache.pulsar.broker.stats.PrometheusMetricsTest)  Time elapsed: 0.207 s  <<< FAILURE!
  java.lang.AssertionError: expected [1.0] but found [2.0]
  	at org.testng.Assert.fail(Assert.java:99)
  	at org.testng.Assert.failNotEquals(Assert.java:1037)
  	at org.testng.Assert.assertEquals(Assert.java:701)
  	at org.testng.Assert.assertEquals(Assert.java:712)
  	at org.apache.pulsar.broker.stats.PrometheusMetricsTest.compareBrokerConnectionStateCount(PrometheusMetricsTest.java:1381)
  	at org.apache.pulsar.broker.stats.PrometheusMetricsTest.testBrokerConnection(PrometheusMetricsTest.java:1321)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
  	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
  	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  	at java.base/java.lang.Thread.run(Thread.java:833)
</code></pre>
</details>


### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/PrometheusMetricsTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/PrometheusMetricsTest.java
@@ -1318,16 +1318,16 @@ public class PrometheusMetricsTest extends BrokerTestBase {
         String metricsStr = statsOut.toString();
         Multimap<String, Metric> metrics = parseMetrics(metricsStr);
         List<Metric> cm = (List<Metric>) metrics.get("pulsar_connection_created_total_count");
-        compareBrokerConnectionStateCount(cm, 1.0);
+        compareBrokerConnectionStateCount(cm, 2.0);
 
         cm = (List<Metric>) metrics.get("pulsar_connection_create_success_count");
-        compareBrokerConnectionStateCount(cm, 1.0);
+        compareBrokerConnectionStateCount(cm, 2.0);
 
         cm = (List<Metric>) metrics.get("pulsar_connection_closed_total_count");
         compareBrokerConnectionStateCount(cm, 0.0);
 
         cm = (List<Metric>) metrics.get("pulsar_active_connections");
-        compareBrokerConnectionStateCount(cm, 1.0);
+        compareBrokerConnectionStateCount(cm, 2.0);
 
         pulsarClient.close();
         statsOut = new ByteArrayOutputStream();
@@ -1365,13 +1365,13 @@ public class PrometheusMetricsTest extends BrokerTestBase {
         compareBrokerConnectionStateCount(cm, 1.0);
 
         cm = (List<Metric>) metrics.get("pulsar_connection_create_success_count");
-        compareBrokerConnectionStateCount(cm, 1.0);
+        compareBrokerConnectionStateCount(cm, 2.0);
 
         cm = (List<Metric>) metrics.get("pulsar_active_connections");
-        compareBrokerConnectionStateCount(cm, 0.0);
+        compareBrokerConnectionStateCount(cm, 1.0);
 
         cm = (List<Metric>) metrics.get("pulsar_connection_created_total_count");
-        compareBrokerConnectionStateCount(cm, 2.0);
+        compareBrokerConnectionStateCount(cm, 3.0);
     }
 
     private void compareBrokerConnectionStateCount(List<Metric> cm, double count) {

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/PrometheusMetricsTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/PrometheusMetricsTest.java
@@ -94,9 +94,6 @@ public class PrometheusMetricsTest extends BrokerTestBase {
     protected void setup() throws Exception {
         conf.setTopicLevelPoliciesEnabled(false);
         conf.setSystemTopicEnabled(false);
-        conf.setTransactionCoordinatorEnabled(true);
-        conf.setTransactionLogBatchedWriteEnabled(true);
-        conf.setTransactionPendingAckBatchedWriteEnabled(true);
         super.baseSetup();
         AuthenticationProviderToken.resetMetrics();
     }
@@ -782,6 +779,12 @@ public class PrometheusMetricsTest extends BrokerTestBase {
     // Running the test twice to make sure types are present when generated multiple times
     @Test(invocationCount = 2)
     public void testDuplicateMetricTypeDefinitions() throws Exception {
+        cleanup();
+        conf.setTransactionCoordinatorEnabled(true);
+        conf.setTransactionLogBatchedWriteEnabled(true);
+        conf.setTransactionPendingAckBatchedWriteEnabled(true);
+        setup();
+
         Set<String> allPrometheusSuffixString = allPrometheusSuffixEnums();
         Producer<byte[]> p1 = pulsarClient.newProducer().topic("persistent://my-property/use/my-ns/my-topic1").create();
         Producer<byte[]> p2 = pulsarClient.newProducer().topic("persistent://my-property/use/my-ns/my-topic2").create();
@@ -861,6 +864,10 @@ public class PrometheusMetricsTest extends BrokerTestBase {
 
         p1.close();
         p2.close();
+
+        conf.setTransactionCoordinatorEnabled(false);
+        conf.setTransactionLogBatchedWriteEnabled(false);
+        conf.setTransactionPendingAckBatchedWriteEnabled(false);
     }
 
     /***
@@ -1318,16 +1325,16 @@ public class PrometheusMetricsTest extends BrokerTestBase {
         String metricsStr = statsOut.toString();
         Multimap<String, Metric> metrics = parseMetrics(metricsStr);
         List<Metric> cm = (List<Metric>) metrics.get("pulsar_connection_created_total_count");
-        compareBrokerConnectionStateCount(cm, 2.0);
+        compareBrokerConnectionStateCount(cm, 1.0);
 
         cm = (List<Metric>) metrics.get("pulsar_connection_create_success_count");
-        compareBrokerConnectionStateCount(cm, 2.0);
+        compareBrokerConnectionStateCount(cm, 1.0);
 
         cm = (List<Metric>) metrics.get("pulsar_connection_closed_total_count");
         compareBrokerConnectionStateCount(cm, 0.0);
 
         cm = (List<Metric>) metrics.get("pulsar_active_connections");
-        compareBrokerConnectionStateCount(cm, 2.0);
+        compareBrokerConnectionStateCount(cm, 1.0);
 
         pulsarClient.close();
         statsOut = new ByteArrayOutputStream();
@@ -1365,13 +1372,13 @@ public class PrometheusMetricsTest extends BrokerTestBase {
         compareBrokerConnectionStateCount(cm, 1.0);
 
         cm = (List<Metric>) metrics.get("pulsar_connection_create_success_count");
-        compareBrokerConnectionStateCount(cm, 2.0);
-
-        cm = (List<Metric>) metrics.get("pulsar_active_connections");
         compareBrokerConnectionStateCount(cm, 1.0);
 
+        cm = (List<Metric>) metrics.get("pulsar_active_connections");
+        compareBrokerConnectionStateCount(cm, 0.0);
+
         cm = (List<Metric>) metrics.get("pulsar_connection_created_total_count");
-        compareBrokerConnectionStateCount(cm, 3.0);
+        compareBrokerConnectionStateCount(cm, 2.0);
     }
 
     private void compareBrokerConnectionStateCount(List<Metric> cm, double count) {

